lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -360,12 +361,15 @@ void ProxyRenderDelegate::_InitRenderDelegate(MSubSceneContainer& container) {
         _selection.reset(new HdSelection);
 
 #if defined(WANT_UFE_BUILD)
-        if (!_ufeSelectionObserver) {
+        if (!_observer) {
+            _observer = std::make_shared<UfeObserver>(*this);
+
             auto globalSelection = Ufe::GlobalSelection::get();
             if (globalSelection) {
-                _ufeSelectionObserver = std::make_shared<UfeSelectionObserver>(*this);
-                globalSelection->addObserver(_ufeSelectionObserver);
+                globalSelection->addObserver(_observer);
             }
+
+            Ufe::Scene::instance().addObjectAddObserver(_observer);
```

Comment:
Does it make sense to add this object add observer even if there is no global selection list?

Reply:
I am not sure. In what scenario there won't be global selection list?

The check of the global selection list was intended to avoid crash. But maybe it is always non-null.

Reply:
Certainly in Maya there will always be a global selection list.  If the global selection list were null, many, many selection-based workflows would either not work or crash.  It is created during Maya initialization, so we could assert that it is non-null.

Reply:
Even for command line render?

Reply:
Chatted with @ppt-adsk and the answer is yes.

Reply:
Are you then going to switch it to TF_CODING_ERROR?

Reply:
Done with the new commit.